assets/app/scripts/directives/nav.js

```diff
@@ -57,11 +78,14 @@ angular.module('openshiftConsole')
             $location.url(newURL);
           });
         });
-        $scope.$watch("projects", function(projects) {
+
+        var clearAndUpdateOptions = function() {
```

Comment:
`$scope.$watch` calls this immediately... according to https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$watch you can detect and ignore this case by doing an `===` comparison on the `newValue`,`oldValue` args and returning early when true. Should probably do that or the selector will be rebuilt three times every time we navigate... I'm sure that doesn't help the flickering.